janusgraph-lucene/src/test/java/org/janusgraph/graphdb/IndexMetricTest.java

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb;
+
+import static org.janusgraph.StorageSetup.getHomeDir;
+import static org.janusgraph.diskstorage.Backend.METRICS_INDEX_PROVIDER_NAME;
+import static org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider.M_MUTATE;
+import static org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider.M_QUERY;
+import static org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider.M_RAW_QUERY;
+import static org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider.M_TOTALS;
+import static org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider.OPERATION_NAMES;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.BASIC_METRICS;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.INDEX_BACKEND;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.INDEX_DIRECTORY;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.INDEX_NS;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.METRICS_PREFIX_DEFAULT;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.janusgraph.BerkeleyStorageSetup;
+import org.janusgraph.core.JanusGraphException;
+import org.janusgraph.core.PropertyKey;
+import org.janusgraph.core.schema.Mapping;
+import org.janusgraph.diskstorage.configuration.BasicConfiguration;
+import org.janusgraph.diskstorage.configuration.ModifiableConfiguration;
+import org.janusgraph.diskstorage.configuration.WriteConfiguration;
+import org.janusgraph.diskstorage.util.MetricInstrumentedIndexProvider;
+import org.janusgraph.diskstorage.util.MetricInstrumentedStore;
+import org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration;
+import org.janusgraph.util.stats.MetricManager;
+
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.Set;
+
+public class IndexMetricTest extends JanusGraphBaseTest {
```

Comment:
Shouldn't this be an abstract class under `janusgraph-backend-testsutils` module, and then extended by `BerkeleyLuceneIndexMetricTest` under `janusgraph-lucene` module?